controlplane/src/core/repositories/UserRepository.ts

```diff
@@ -85,4 +85,24 @@ export class UserRepository {
       await orgRepo.addOrganizationMemberRoles({ memberID: insertedMember.id, roles: ['member'] });
     });
   }
+
+  // checks if the user has the right roles to perform the operation
+  public async checkUserAccess(input: { userID?: string; organizationID: string; rolesToBe: MemberRole[] }) {
```

Comment:
As described earlier let's put this logic in a separate service or centralize it with the authContext logic.

Reply:
Done